### DISCUSSION TITLE
Oqtane 5.0 / .NET 8 Release Decision

### DISCUSSION THREAD

**Discussion by sbwalker** [oqtane/oqtane.framework#3453] [General]
It was discussed at the Developer Meeting yesterday and the consensus was that we should release Oqtane 5.0 on .NET 8 with support for only the legacy interactive hosting models at this time. This will allow people to use Oqtane on .NET 8 with the confidence that it is backward compatible and all scenarios work as expected. 

We will continue to explore the migration to the new Blazor SSR approach – mainly by validating scenarios through the OqtaneSSR POC project (https://github.com/oqtane/OqtaneSSR). At this point there are enough blockers that it does not make sense to attempt a migration. However would be great if we could offer this capability in an Oqtane 5.1 release in Q1 2024… so we definitely will continue pushing forward.

**Comment by sbwalker** [comment]
I will merge the "net8" branch into the "dev" branch early next week as we prepare for the 5.0 release.

**Comment by sbwalker** [comment]
The "net8" branch has now been merged into the "dev" branch. This means that in order to use the "dev" branch you will need to upgrade your environment to the Visual Studio 2022 Preview  release and install .NET 8. The official .NET 8 release will be available next week - which will require updating all of the package references to the final .NET 8 versions. To keep developing modules or themes for older versions of Oqtane/.NET, please see this blog: https://www.oqtane.org/blog/!/73/developing-on-legacy-versions-of-oqtane-and-net

**Comment by sbwalker** [comment]
.NET 8 is being released tomorrow, Nov 14... Oqtane 5.0 will be released on Thursday, Nov 16.

**Comment by sbwalker** [comment]
#3475 includes updates to the official .NET 8 packages

**Reply by sbwalker** [reply]
Note that there are a number of packages related to database dependencies which do not appear to have official .NET 8 packages available yet:

```
<PackageReference Include="MySql.EntityFrameworkCore" Version="8.0.0-preview" />
<PackageReference Include="EFCore.NamingConventions" Version="8.0.0-rc.2" />
<PackageReference Include="Npgsql.EntityFrameworkCore.PostgreSQL" Version="8.0.0-rc.2" />
<PackageReference Include="Microsoft.Data.SqlClient" Version="5.2.0-preview3.23201.1" />
<PackageReference Include="SQLitePCLRaw.bundle_e_sqlite3" Version="2.1.7-pre20231110210158" />
```

**Comment by sbwalker** [comment]
I have successfully upgraded 4 different App Services on Azure to Oqtane 5.0 on .NET 8...

First I had to stop the app service and set the .NET Version to 8.0 LTS:

![image](https://github.com/oqtane/oqtane.framework/assets/4840590/08af2e49-101d-4ea0-93a6-e1fd89080c32)

Then I used Kudu in the Advanced Tools to upload the Upgrade.zip package. Once finished I restarted the app service and tried to browse to the site and but received an error message:

![image](https://github.com/oqtane/oqtane.framework/assets/4840590/e5832958-2c78-4848-a24c-397382589e88)

So I did a Restart on the app service and waited 10-20 seconds. I then tried to browse again and I got a different message:

![image](https://github.com/oqtane/oqtane.framework/assets/4840590/3c009cd6-0507-4c8b-ad69-3ee3cfbcf92b)

which basically meant the site was still restarting... so with some browser refreshes and patience the site eventually  came up

![image](https://github.com/oqtane/oqtane.framework/assets/4840590/570c8d8c-9af6-4b80-ace1-d9afdb56db65)

All signs look positive for a release of 5.0 tomorrow.

**Reply by ADefWebserver** [reply]
I had the exact same experience with:
https://oqtane-blazor.azurewebsites.net/

- Updated Azure to .Net8
- Used KUDU to unzip the Upgrade package
- Got the same errors. Had to wait then it worked

The site loads, I log in as host, then I get:

![image](https://github.com/oqtane/oqtane.framework/assets/1857799/4e6a9408-6d24-4048-bf56-95a646d1b334)

If I open another web browser the site will come up and I will be logged in.

**Reply by ADefWebserver** [reply]
For now I just re-installed the site. I suspect that I had Azure B2C login configured messed up the upgrade.

**Comment by sbwalker** [comment]
Blazor got a LOT larger in .NET 8 from a distribution perspective:

![image](https://github.com/oqtane/oqtane.framework/assets/4840590/0a768f03-432a-45df-a02a-350bd0c6809d)

**Comment by sbwalker** [comment]
@leigh-pointer the plan was to release Oqtane 5.0 today. I am unable to reproduce any of the issues you logged yesterday in either a local development environment using Visual Studio, a local IIS deployment,  or production Azure deployment. So I am inclined to proceed with the release... what are your thoughts?

**Comment by sbwalker** [comment]
.NET 8 includes a lot of new localization folders when you build the solution. It appears that this is for some CodeAnalysis translations. Each of these folders consumes ~500KB of space - which adds ~7MB to a .NET 8 release. I am wondering if these folders could be removed.

![image](https://github.com/oqtane/oqtane.framework/assets/4840590/adc69120-be9f-4331-9a65-04ba304d83bc)

These satellite assemblies also seem to create some side effects for Oqtane, as I see some messages such as:

[Error] [Oqtane.Controllers.InstallationController] The Satellite Assembly Folder For de Does Not Exist 

in my error.log when running on WebAssembly.

**Reply by sbwalker** [reply]
This stack overflow post contains more information on this topic:

https://stackoverflow.com/questions/58668618/how-to-stop-the-localized-microsoft-codeanalysis-resources-dll-files-from-gett

**Comment by sbwalker** [comment]
The following 2 packages now have official releases for .NET 8

```
<PackageReference Include="Npgsql.EntityFrameworkCore.PostgreSQL" Version="8.0.0" />
<PackageReference Include="SQLitePCLRaw.bundle_e_sqlite3" Version="2.1.7" />
```
